### ISSUE TITLE
Gpoddersync only processes first EpisodeAction

### ISSUE THREAD

**Issue by JonOfUs** [thrillfall/nextcloud-gpodder#20]
episode_action/create only deals with the first EpisodeAction it finds inside the data string.

https://github.com/thrillfall/nextcloud-gpodder/blob/ab155ca94ac0b7c3106c8d1bc1e6968500fe3427/lib/Controller/EpisodeActionController.php#L56-L58

Nevertheless, AntennaPod clearly sends a list of possibly multiple EpisodeActions:
`String body = createBody(queuedEpisodeActions.toString());`

This could be a problem when synchronizing AntennaPod for the first time, because then only the first EpisodeAction will be stored.

Can you confirm that this could lead to problems? I am not sure about it, it seems like a too obvious bug, Could be on the wrong track here.

**Comment by thrillfall** [comment]
You are very right about this. Now I remember I wanted to fix this.

I leave it to you to fix this if you want to.
If you do please add a test for processing multiple episode actions. Also please take the opportunity to extract the logic from the controller.

If you need any further help please just ask

**Comment by JonOfUs** [comment]
On the other hand, even during the first synchronization AntennaPod only sends a single EpisodeAction. It seems like the app never sends more, even on full sync and with lots of unsynced episode changes. So could this be an AntennaPod issue too?

**Comment by thrillfall** [comment]
No this is not a bug. Its how antennapod synchronizes episode actions. 
You can "gather" multiple episode actions by disabling mobile updates in the network settings. Then playing some episodes and then reenable mobile updates.
Antennapod will then send a list of episode actions to the server

**Comment by JonOfUs** [comment]
You're right, after disabling internet multiple EpisodeActions got sent.
What I tried was disabling synchronization (by logging out), then changing some episode states and then re-enabling synchronization. Then only one EpisodeAction will be synchronized, and it will be none of the episode state changes that I did while sync was disabled.

So it seems like the synchronization works only on episode state changes that are carried out while the sync is on. Probably because the EpisodeActions queue won't be filled when sync is off or something. This, however, disables to send EpisodeActions from before sync existed or, in my example, from the time in which sync was disabled.
